core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -181,8 +208,39 @@ class CudaTeamMember {
   }
 
   template <class Closure, class ValueType>
-  KOKKOS_INLINE_FUNCTION void team_broadcast(Closure const& f, ValueType& val,
-                                             const int& thread_id) const {
+  KOKKOS_INLINE_FUNCTION
+      typename std::enable_if<(sizeof(ValueType) < 4), void>::type
+      team_broadcast(Closure const& f, ValueType& val,
+                     const int& thread_id) const {
+    using conv_type = uint32_t;
```

Comment:
Same here.

Reply:
Not sure I understand your comment.  Do you mean, the `using conv_type = uint32_t;` or the whole function signature?  

Because the specialization for team_broadcast separates the types using a boolean expression `size>=4` and `size < 4` the function signature has to match for all 4 overloads (2 with functor and 2 without).  If I put the function inside an ifdef, there will be a potential mismatch.

Reply:
Just the `using` declaration. `CUDA-9.2-Clang` is complaining.